test/test_mixed_function_space_with_mixed_mesh.py

```diff
@@ -0,0 +1,81 @@
+from ufl import (
+    CellVolume,
+    Coefficient,
+    FacetArea,
+    FacetNormal,
+    FunctionSpace,
+    Measure,
+    Mesh,
+    MixedMesh,
+    SpatialCoordinate,
+    TestFunction,
+    TrialFunction,
+    grad,
+    inner,
+    split,
+    triangle,
+)
+from ufl.algorithms import compute_form_data
+from ufl.domain import extract_domains
+from ufl.finiteelement import FiniteElement, MixedElement
+from ufl.pullback import contravariant_piola, identity_pullback
+from ufl.sobolevspace import H1, L2, HDiv
+
+
+def test_mixed_function_space_with_mixed_mesh_basic():
+    cell = triangle
+    elem0 = FiniteElement("Lagrange", cell, 1, (), identity_pullback, H1)
+    elem1 = FiniteElement("Brezzi-Douglas-Marini", cell, 1, (2,), contravariant_piola, HDiv)
+    elem2 = FiniteElement("Discontinuous Lagrange", cell, 0, (), identity_pullback, L2)
+    elem = MixedElement([elem0, elem1, elem2])
+    mesh0 = Mesh(FiniteElement("Lagrange", cell, 1, (2,), identity_pullback, H1), ufl_id=100)
+    mesh1 = Mesh(FiniteElement("Lagrange", cell, 1, (2,), identity_pullback, H1), ufl_id=101)
+    mesh2 = Mesh(FiniteElement("Lagrange", cell, 1, (2,), identity_pullback, H1), ufl_id=102)
+    domain = MixedMesh(mesh0, mesh1, mesh2)
+    V = FunctionSpace(domain, elem)
```

Comment:
So we now have two ways of creating a `FunctionSpace` with `MixedElement`s.
1. A single mesh of class `Mesh` with a `MixedElement`, where `MixedElement` can have arbitrary many elements
2. A `MixedMesh` consisting of `M` meshes, and a `MixedElement` consisting of `M` elements.

Do we need additional checks in the `FunctionSpace` constructor to ensure that this is satisfied (instead of catching this at a later instance, inside for instance `GenericDerivativeRuleset.reference_value`.

I would also like to highlight that the API for `MixedElement` and `MixedMeshes` differs, as one passes a list, while the other unrolls a list, I think we should be consistent, and use lists in both places

Reply:
Added checks in `FunctionSpace` constructor and changed `MixedMesh` API as suggested.  Thanks.